force-app/main/default/lwc/geGatewaySettings/geGatewaySettings.js

```diff
@@ -1,20 +1,52 @@
 import { registerListener } from 'c/pubsubNoPageRef';
+import decryptGatewayId from '@salesforce/apex/GE_GiftEntryController.decryptGatewayId';
+import { showToast } from 'c/utilCommon';
+import psGatewayNotValid from '@salesforce/label/c.psGatewayNotValid';
 
 class GeGatewaySettings {
 
     elevateSettings = {};
     templateRecordId = null;
+    decryptedGatewayId = null;
+    isGiftEntryBatch = false;
 
     setElevateSettings(initialSettings, templateRecordId) {
+        this.isGiftEntryBatch = false;
         this.elevateSettings = initialSettings;
         this.templateRecordId = templateRecordId;
         registerListener('updateElevateSettings', this.settings, this);
     }
 
+    initDecryptedElevateSettings(elevateSettings) {
+        this.isGiftEntryBatch = true;
+        this.elevateSettings = elevateSettings;
+        if (this.elevateSettings?.uniqueKey) {
+            this.decryptElevateGateway().then(error => {
+                if (this.elevateSettings?.uniqueKey && !this.decryptedGatewayId) {
+                    showToast(psGatewayNotValid, details, 'error', 'sticky');
```

Comment:
Where are we getting this `details` property from? Also we should be using the `catch` method to catch the potential error from this promise

Reply:
details is passed from handleErrors.

Reply:
I'm wrong. I'll take care of that. Copy/paste error...